src/java.base/share/classes/sun/security/ssl/ServerHello.java

```diff
@@ -431,9 +433,11 @@ private static KeyExchangeProperties chooseCipherSuite(
 
                 SSLKeyExchange ke = SSLKeyExchange.valueOf(
                         cs.keyExchange, shc.negotiatedProtocol);
+
                 if (ke == null) {
                     continue;
```

Comment:
I may add a debug log here that the key exchange is not good for the cipher suite and protocol.

Reply:
As mentioned in my previous reply, if handshaking failed, all of the cipher suites will be logged. Technically key name is  included in cipher suite name.

Reply:
Hm, all cipher suites are logged, please analysis the failure by yourself.  Is it really the purpose of debug log and what you want to do?